src/macos/mod.rs

```diff
@@ -37,12 +37,12 @@ impl<'a> SegmentTrait for Segment<'a> {
     type SharedLibrary = ::macos::SharedLibrary<'a>;
 
     #[inline]
-    fn name(&self) -> &OsStr {
+    fn name(&self) -> &str {
         let cstr = match *self {
             Segment::Segment32(seg) => unsafe { CStr::from_ptr(seg.segname.as_ptr()) },
             Segment::Segment64(seg) => unsafe { CStr::from_ptr(seg.segname.as_ptr()) },
         };
-        OsStr::from_bytes(cstr.to_bytes())
+        cstr.to_str().unwrap_or("")
```

Comment:
Please use something else as fallback to make it easier to figure out what is going on when the segment name is not UTF-8.

Reply:
Taking suggestions. Realistically this should not happen however as the file format specifies this to be UTF-8.

Reply:
Actually we have a similar fallback already on linux, so i added one that matches somewhat.